drivers/modem/Kconfig

```diff
@@ -30,15 +30,66 @@ config MODEM_RECEIVER
 	  in that callbacks are executed in a different work queue and data is
 	  passed around in k_pipe structures.
 
+if MODEM_RECEIVER
+
 config MODEM_RECEIVER_MAX_CONTEXTS
 	int "Maximum number of modem receiver contexts"
-	depends on MODEM_RECEIVER
 	range 1 10
 	default 1
 	help
 	  Maximum number of modem receiver contexts to handle.  For most
 	  purposes this should stay at 1.
 
+endif # MODEM_RECEIVER
+
+config MODEM_CONTEXT
+	bool "Enable modem context helper driver"
+	help
+	  This driver allows modem drivers to communicate with an interface
+	  using custom defined protocols. Driver doesn't inspect received data
+	  and all aspects of received protocol data are handled by application
+	  work method provided.  This driver combines abstractions for:
+	  modem interface, command handler, pin config and socket handling.
+
+if MODEM_CONTEXT
+
+config MODEM_CONTEXT_MAX_NUM
+	int "Maximum number of modem contexts"
+	default 1
+	help
+	  Maximum number of modem contexts to handle.  For most
+	  purposes this should stay at 1.
+
+config MODEM_CONTEXT_VERBOSE_DEBUG
+	bool "Enable verbose debug in modem context"
+	help
+	  Enabling this setting will turn on VERY heavy debugging from the
+	  modem context helper.  Only to be used for debugging.
+
+config MODEM_IFACE_UART
+	depends on SERIAL_SUPPORT_INTERRUPT
+	select UART_INTERRUPT_DRIVEN
+	select RING_BUFFER
+	bool "Enable UART-based modem interface"
+
+config MODEM_CMD_HANDLER
+	bool "Enable generic modem command handler"
```

Comment:
Some help texts for these might be nice, to quickly point people to what to do next after they enable them.